docs/api/migration-guides/qiskit-1.0-features.mdx

```diff
@@ -0,0 +1,1003 @@
+The `qiskit.converters.ast_to_dag` function is deprecated. It converted the
+abstract syntax tree generated by the legacy OpenQASM 2 parser to a
+`DAGCircuit`. As the legacy OpenQASM 2 parser has been removed (see
+[qiskit.qasm](#qiskit.qasm)), this function no longer serves a purpose.
+Instead, parse your OpenQASM 2 files into a `QuantumCircuit` using the
+[`QuantumCircuit.from_qasm_file`](/api/qiskit/qiskit.circuit.QuantumCircuit#from_qasm_file)
+or
+[`QuantumCircuit.from_qasm_str`](/api/qiskit/qiskit.circuit.QuantumCircuit#from_qasm_str)
+constructor methods (or the [`qiskit.qasm2`](/api/qiskit/qasm2) module),
+then convert that `QuantumCircuit` into a
+[`DAGCircuit`](/api/qiskit/qiskit.dagcircuit.DAGCircuit) with
+[`circuit_to_dag`](/api/qiskit/converters#circuit_to_dag).
+
+```python
+# Deprecated
+from qiskit.converters import ast_to_dag
+from qiskit.qasm import Qasm
+
+dag = ast_to_dag(Qasm(filename="myfile.qasm").parse())
+
+# Alternative
+import qiskit.qasm2
+from qiskit.converters import circuit_to_dag
+
+dag = circuit_to_dag(qiskit.qasm2.load("myfile.qasm"))
+```
+
+<span id="qiskit.extensions"></span>
+## qiskit.extensions
+
+The `qiskit.extensions` module is no longer available. Most of its objects have been
+integrated into the circuit library
+([`qiskit.circuit.library`](/api/qiskit/circuit_library)). To migrate to the
+new location, simply replace `qiskit.extensions` with `qiskit.circuit.library`
+in the object import path. This is a drop-in replacement.
+
+```python
+# Deprecated
+from qiskit.extensions import DiagonalGate
+
+# Alternative
+from qiskit.circuit.library import DiagonalGate
+```
+
+The classes moved to [`qiskit.circuit.library`](/api/qiskit/circuit_library) are:
+
+* [`DiagonalGate`](/api/qiskit/qiskit.circuit.library.DiagonalGate)
+* [`HamiltonianGate`](/api/qiskit/qiskit.circuit.library.HamiltonianGate)
+* [`Initialize`](/api/qiskit/qiskit.circuit.library.Initialize)
+* [`Isometry`](/api/qiskit/qiskit.circuit.library.Isometry)
+* `qiskit.circuit.library.generalized_gates.mcg_up_diag.MCGupDiag`
+* [`UCGate`](/api/qiskit/qiskit.circuit.library.UCGate)
+* [`UCPauliRotGate`](/api/qiskit/qiskit.circuit.library.UCPauliRotGate)
+* [`UCRXGate`](/api/qiskit/qiskit.circuit.library.UCRXGate)
+* [`UCRYGate`](/api/qiskit/qiskit.circuit.library.UCRYGate)
+* [`UCRZGate`](/api/qiskit/qiskit.circuit.library.UCRZGate)
+* [`UnitaryGate`](/api/qiskit/qiskit.circuit.library.UnitaryGate)
+
+The following classes have been removed from the codebase, as their
+functionality was either redundant or linked to the `extensions` module:
+
+Deprecated                       |   Alternative
+---------------------------------|--------------------------------------------
+[`SingleQubitUnitary`](/api/qiskit/0.46/qiskit.extensions.SingleQubitUnitary)    |  [`qiskit.circuit.library.UnitaryGate`](/api/qiskit/qiskit.circuit.library.UnitaryGate)
+[`Snapshot`](/api/qiskit/0.46/qiskit.extensions.Snapshot)             |  use qiskit-aer's save instructions
```

Comment:
Is there a way that we can link to the qiskit-aer API docs? (not essential but might be good to add in a follow-up)

Reply:
Good idea, yes we can definitely do that

Reply:
Done in 75b20c10557e32f56c16f4b99246de6371df5552